pxd.c

```diff
@@ -937,12 +946,37 @@ void pxd_reroute_slowpath(struct request_queue *q, struct bio *bio)
 	fuse_request_send_nowait(&pxd_dev->ctx->fc, req);
 }
 
+
+#ifdef __PXD_BIO_BLKMQ__
 #if !defined(__PX_BLKMQ__)
+void pxdmq_reroute_slowpath(struct fuse_req *req)
+{
+    struct pxd_device *pxd_dev = req->pxd_dev;
+    struct request *rq = req->rq;
+
+    BUG_ON(pxd_dev->fp.fastpath);
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0) || defined(REQ_PREFLUSH)
+    if (pxd_request(req, blk_rq_bytes(rq), blk_rq_pos(rq) * SECTOR_SIZE,
+                  pxd_dev->minor, req_op(rq), rq->cmd_flags)) {
+#else
+    if (pxd_request(req, blk_rq_bytes(rq), blk_rq_pos(rq) * SECTOR_SIZE,
+                  pxd_dev->minor, rq->cmd_flags)) {
+#endif
+        fuse_request_free(req);
+        blk_end_request(rq, -EIO, blk_rq_bytes(rq));
+        return;
+    }
+    fuse_request_send_nowait(&pxd_dev->ctx->fc, req);
+}
+
+
 static void pxd_rq_fn(struct request_queue *q)
```

Comment:
This works but I think it would be good to separate out native and fastpath version of the fn. And register with a wrapper which can redirect call to native/fastpath. The reason to do this is that any subsequent fastpath related changes would not touch native fn().

Reply:
It already is like that, the guts of the logic for fastpath processing is moved to pxd_bio_{makereq|blkmq} files. The common entry function, just redirects IO if fastpath enabled.